src/workload/mod.rs

```diff
@@ -373,27 +384,44 @@ pub enum Component {
 
 #[derive(Clone)]
 pub struct Topics {
+    protocol: Protocol,
     topics: Vec<Arc<String>>,
+    partitions: usize,
     topic_dist: Distribution,
+    partition_dist: Distribution,
+    key_len: usize,
     message_len: usize,
     subscriber_poolsize: usize,
     subscriber_concurrency: usize,
+    publisher_poolsize: usize,
+    publisher_concurrency: usize,
 }
 
 impl Topics {
     pub fn new(config: &Config, topics: &config::Topics) -> Self {
         // ntopics must be >= 1
+        let protocol = config.general().protocol();
```

Comment:
Why is protocol included here? You've split the comment re: ntopics from its declaration by putting this here too

Reply:
Because generate_pubsub uses the protocol to decide whether sending out a KafkaMessage or  PublisherWorkItem::Publish